guides/operator/custom-boms.md

```diff
@@ -0,0 +1,213 @@
+configuration of a service are captured with a release, and don't require user
+intervention when installing Spinnaker with Halyard (or any other tooling that
+takes advantage of this configuration). This cannonical configuration is
+captured from each service's `./halconfig` directory in each repository during
+a build. [Here are Clouddriver's latest default configuration files, as an
+example](https://github.com/spinnaker/clouddriver/tree/master/halconfig).
+
+BOMs can be stored in a number of places, but regardless of the location or
+storage solution, the required configuration files all share the same
+directory/naming conventions relative to a `${CONFIG_INPUT_ROOT}`.
+
+The conventions are as follows:
+
+```
+${CONFIG_INPUT_ROOT}/
+├── boms/
+│   └── ${VERSION}.yml            # for each top-level spinnaker version
+└── ${SUBCOMPONENT}
+    ├── ${DEFAULT_PROFILE}        # when a given version isn't found
+    └── ${SUBCOMPONENT_VERSION}/  # for each subcomponent version
+        └── ${PROFILE}            # for each profile at this version
+```
+
+Take for example the public `gs://halconfig` bucket, where all releases are
+published, looks like (with a lot of omissions):
+
+```
+gs://halconfig/
+├── boms/
+│   ├── 1.10.1.yml
+│   ├── 1.10.0.yml
+│   ├── 1.9.5.yml
+│   ├── 1.9.4.yml
+│   ├── master-latest-unvalidated.yml
+│   └── ...
+├── clouddriver/
+│   ├── clouddriver.yml
+│   ├── 4.0.1-20181024113115/
+│   │   ├── clouddriver.yml
+│   │   ├── clouddriver-caching.yml
+│   │   └── clouddriver-ro.yml
+│   ├── 4.0.0-20181005152808/
+│   │   ├── clouddriver.yml
+│   │   ├── clouddriver-caching.yml
+│   │   └── clouddriver-ro.yml
+│   └── ...
+├── deck/
+│   ├── 2.5.1-20181018042808/
+│   │   └── settings.js
+│   ├── 2.5.1-20181018042808/
+│   │   └── settings.js
+│   └── ...
+└── ...
+```
+
+### BOMS and Configuration in GCS
+
+By default, all BOMs are stored in a publicly readable GCS bucket,
+`gs://halconfig` that Halyard reads from when doing a deployment. This bucket
+location can be changed by setting the `spinnaker.config.input.bucket` property
+in `/opt/spinnaker/config/halyard-local.yml`. If you supply a private bucket,
+Halyard will use your [application default
+credentials](https://cloud.google.com/docs/authentication/production) to
+authenticate.
+
+Given any GCS bucket, `gs://${BUCKET}`, the `${CONFIG_INPUT_ROOT}` is
+`gs://${BUCKET}`. 
+
+As a result, a BOM at version `${VERSION}` can be found at
+`gs://${BUCKET}/boms/${VERSION}.yml`, and a configuration file `${PROFILE}` for
+subcomponent `${SUBCOMPONENT}` at version `${SUBCOMPONENT_VERSION}` can be
+found at `gs://${BUCKET}/${SUBCOMPONENT}/${SUBCOMPONENT_VERSION}/${PROFILE}`.
+
+#### Disabling GCS reads
+
+You can also completely disable reads from GCS by setting
+`spinnaker.config.input.gcs.enabled: false` in
+`/opt/spinanker/config/halyard-local.yml`. Be sure to restart the Halyard
+daemon for this configuration to take effect: `hal shutdown && hal`.
```

Comment:
What's the effect of disabling read from GCS? Is this meant to go hand-in-hand with the next section, configuration on your filesystem?

Reply:
No, you can have both reads from GCS as well as read from the filesystem. The below section describes that with "the configuration files will by default be read from GCS, unless..."

Reply:
For a little more context, the ability to load from the filesystem has two purposes: 1. like you pointed out, to not require GCS, and 2. to make it easy to configure custom bills of materials (even if some shared resources live in the cloud).